src/lib/y2storage/dialogs/guided_setup.rb

```diff
@@ -65,8 +65,25 @@ def run
           "select_filesystem" => { next: :next, back: :back,  abort: :abort }
         }
 
+        # Skip disks selection when there is only one candidate
+        if candidate_disks.size == 1
+          settings.candidate_devices = candidate_disks.map(&:name)
+          sequence["ws_start"] = "select_root_disk"
+        end
+
         Yast::Sequencer.Run(aliases, sequence)
       end
+
+      # Disks that could be selected for installation.
+      #
+      # @return [Array<Disk>]
+      def candidate_disks
+        return @candidate_disks if @candidate_disks
+        candidates = settings.candidate_devices || []
+        candidates = candidates.map { |d| analyzer.device_by_name(d) }
+        candidates = analyzer.candidate_disks if candidates.empty?
```

Comment:
I am not sure if I get logic there. If settings is set, it will use it. And if settings is not set, it will return result of analyzer. So if you going back in guided setup and you previously pick only one disk, it will skip selecting disks, right?